mmpose/models/heads/vis_heads/vis_head.py

```diff
@@ -0,0 +1,217 @@
+
+        self.pose_head = MODELS.build(pose_cfg)
+        self.pose_cfg = pose_cfg
+
+        self.vis_head = nn.Sequential(
+            nn.AdaptiveAvgPool2d(1), nn.Flatten(),
+            nn.Linear(self.in_channels, self.num_joints), nn.Sigmoid())
+
+    def forward(self, feats: Tuple[Tensor]):
+        """Forward the network. The input is multi scale feature maps and the
+        output is coordinates visibility.
+
+        Args:
+            feats (Tuple[Tensor]): Multi scale feature maps.
+
+        Returns:
+            Tensor: output coordinates visibility.
+        """
+        x = feats[-1]
+        while len(x.shape) < 4:
+            x.unsqueeze_(-1)
+        x = self.vis_head(x)
+        return x.reshape(-1, self.num_joints)
+
+    def integrate(self, batch_vis: Tensor,
+                  pose_preds: Union[Tuple, Predictions]) -> InstanceList:
+        """Add keypoints visibility prediction to pose prediction.
+
+        Args:
+            batch_vis (torch.Tensor): predicted batch keypoints visibility
+                in shape (B, num_instances, K, D) where K is the keypoint
+                number, D is the keypoint dimension and num_instances=1,
+                since vis_head only support top-down methods now
+            pose_preds (Union[Tuple, Predictions]): output of keypoints
+                prediction head
+
+        Returns:
+            Union[InstanceList | Tuple[InstanceList | PixelDataList]]: If
+            posehead's ``test_cfg['output_heatmap']==True``, return both
+            pose and heatmap prediction; otherwise only return the pose
+            prediction.
+
+            The pose prediction is a list of ``InstanceData``, each contains
+            the following fields:
+
+                - keypoints (np.ndarray): predicted keypoint coordinates in
+                    shape (num_instances, K, D) where K is the keypoint number
+                    and D is the keypoint dimension
+                - keypoint_scores (np.ndarray): predicted keypoint scores in
+                    shape (num_instances, K)
+                - keypoint_visibility (np.ndarray): predicted keypoints
+                    visibility in shape (num_instances, K)
+
+            The heatmap prediction is a list of ``PixelData``, each contains
+            the following fields:
+
+                - heatmaps (Tensor): The predicted heatmaps in shape (K, h, w)
+        """
+        if isinstance(pose_preds, tuple):
+            pose_pred_instances, pose_pred_fields = pose_preds
+        else:
+            pose_pred_instances = pose_preds
+            pose_pred_fields = None
+
+        batch_vis_np = to_numpy(batch_vis, unzip=True)
+
+        preds = [
+            InstanceData(
+                keypoints=pose_pred_instance.keypoints,
+                keypoint_scores=pose_pred_instance.keypoint_scores,
+                keypoint_visibility=vis_pred) for pose_pred_instance, vis_pred
+            in zip(pose_pred_instances, batch_vis_np)
+        ]
```

Comment:
I think it would be better to directly assign `keypoint_visibility` to `pose_pred_instances` using `pose_pred_instances.keypoint_visibility=vis_pred`, instead of creating new `InstanceData` objects

Reply:
Thanks for your advice. I will rewrite this soon.

Reply:
> I think it would be better to directly assign `keypoint_visibility` to `pose_pred_instances` using `pose_pred_instances.keypoint_visibility=vis_pred`, instead of creating new `InstanceData` objects

I apologize for mistakenly using "git push -f" to commit. Would you please tell me how to fix this mistake. Thanks.

Reply:
@Billccx Just feel free to close this pr and create a new one